### DISCUSSION TITLE
I want to give navbar of landing page some special style. But I don't know how to do it.

### DISCUSSION THREAD

**Discussion by Goddywu** [facebook/docusaurus#5288] [Q&A] [answered]
I have tried to use `useEffect` in `/src/page/index.js`.

```
  useEffect(() => {
    let rootDom = document.getElementById('__docusaurus')
    if (rootDom) rootDom.classList.toggle('homepage')
    return () => {
      if (rootDom) rootDom.classList.toggle('homepage')
    }
  }, [])
```

But this code will execute in some seconds.

first it show like this👇
![image](https://user-images.githubusercontent.com/29056050/128287618-27845d61-0d90-4d95-bc6a-baad28d920eb.png)

after seconds, then change to my expected 
![image](https://user-images.githubusercontent.com/29056050/128287707-44c1e9b4-3d6e-4a05-b717-1704089888bf.png)

Is there a better way to give homepage special style when the very initial?

**Comment by slorber** [comment, marked as answer]
Hi

Using `useEffect`, the static HTML is already rendered to the user when your homepage class is inserted by the effect.
It is not a good fit because it will create some fouc / layout shift issues as the navbar will be first visible and then disappear after React hydration.

One solution to apply a className to the html/body tags on a per-page basis is to use the `Head` component (behind the scene it's using https://github.com/nfl/react-helmet)

```jsx
import React from 'react';
import Head from '@docusaurus/Head';

export default function HomePage() {
  return (
    <>
      <Head>
        <body className="homepage" />
      </Head>
      <Layout>
         <h1>My HomePage</h1>
      </Layout>
    </>
  );
}
```

The homepage class would then be applied reliably on both the client + server, avoiding FOUC issues:

![image](https://user-images.githubusercontent.com/749374/128324583-58433d88-f54b-4557-99d8-9bcc821825c1.png)

---

Another possibility is to swizzle the Navbar component, so that you can add logic to avoid rendering it for certain URL paths.

https://docusaurus.io/docs/using-themes#for-site-owners

Something like this could work (didn't try):

```js
import React from 'react';
import {useLocation} from '@docusaurus/router';
import OriginalNavbar from '@theme-original/Navbar';

export default function Navbar(props) {
  const {pathname} = useLocation();
  if (pathname === "/") {
    return null; 
  }
  return (
    <OriginalNavbar {...props} />
  );
}
```

**Reply by Goddywu** [reply]
hi, slorber

I have use the first solution. It works well!!  

Many thx. XD
